core/chains/chains.go

```diff
@@ -0,0 +1,131 @@
+package chains
+
+import (
+	"math/big"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+// To add a new chain to this file:
+// 1. Add the global var in the vars list
+// 2. Add the chain ID in the map in the init() function
+// 3. Add a config set in configs.go
+
+// Chain represents a blockchain with a unique Chain ID
+type Chain struct {
+	id      *big.Int
+	config  ChainSpecificConfig
+	logOnce sync.Once
+}
+
+func (c *Chain) setChainID(id int64) {
+	c.id = big.NewInt(id)
+}
+
+func (c *Chain) ID() *big.Int {
+	return c.id
+}
+
+func (c *Chain) Config() ChainSpecificConfig {
+	if !c.config.set {
+		c.logOnce.Do(func() {
+			logger.Warnf("chain with ID %s does not have a chain-specific config, using fallback config instead", c.ID())
+		})
+		return FallbackConfig
+	}
+	return c.config
+}
+
+// IsArbitrum returns true if the chain is arbitrum mainnet or testnet
+func (c *Chain) IsArbitrum() bool {
+	if c == nil {
+		return false
+	}
+	return c.ID().Cmp(ArbitrumMainnet.ID()) == 0 || c.ID().Cmp(ArbitrumRinkeby.ID()) == 0
+}
+
+// IsOptimism returns true if the chain is optimism mainnet or testnet
+func (c *Chain) IsOptimism() bool {
+	if c == nil {
+		return false
+	}
+	return c.ID().Cmp(OptimismMainnet.ID()) == 0 || c.ID().Cmp(OptimismKovan.ID()) == 0
+}
+
+var chains map[int64]*Chain
```

Comment:
Why a pointer?

Reply:
It needs to be a singleton because we set the ID and config later, so I wanted to avoid copying.

Maybe there's a way to do this without making it a pointer? I will check.

Reply:
Just checked, yeah it has to be a pointer otherwise `EthMainnet` will not be the same exact object as `ChainByID(1)`